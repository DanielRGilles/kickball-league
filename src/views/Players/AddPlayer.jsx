import { useState, useEffect } from 'react'
import { useHistory } from 'react-router-dom'
import { createPlayer } from'../../services/players.jsx'
import { getTeams } from '../../services/teams.jsx'

export default function AddPlayer() {
    const [ name, setName ] = useState()
    const [ position, setPosition ] = useState()
    const [ teamId, setTeamId ] = useState()
    const history = useHistory();
    const [ teams, setTeams ] = useState([]);

    const loadTeams = async () => {
        const resp = await getTeams();
        setTeams(resp);
      };

    useEffect( () => {
            loadTeams();
        }, [] )

    const handleSubmit = async (e) => {
        e.preventDefault();
        const resp = await createPlayer( { name, position, teamId});
         history.push(`/players/${resp[0].id}`)
    }
    return (
        <>
         <fieldset>
        <legend>Add Player</legend>     
        <form onSubmit={handleSubmit}>
            <label htmlFor='name' >Name:</label>
            <input id='name' name='name' type="text" value={name} onChange={(e) => setName(e.target.value)} />
            <label htmlFor='position' >position:</label>
            <input id='position' name='position' type="text" value={position} onChange={(e) => setPosition(e.target.value)} />
            <label htmlFor='teamId' >teamId:</label>
            <select id='teamId' name='teamId' onChange={(e) => setTeamId(e.target.value)} >
                    {teams.map((team) => <option key={team.id} value={team.id}>{team.name}</option>)}
            </select>
           
            <button type='submit'>Add Player </button>
        </form>
        </fieldset>   
        </>
    )
}
