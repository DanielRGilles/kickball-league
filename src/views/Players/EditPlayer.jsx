import { useState, useEffect } from 'react'
import { useHistory, useParams } from 'react-router-dom'
import { updatePlayerById } from'../../services/players.jsx'
import { getTeams } from '../../services/teams.jsx'

export default function EditPlayer() {
    const [ name, setName ] = useState()
    const [ position, setPosition ] = useState()
    const [ teamId, setTeamId ] = useState()
    const history = useHistory();
    const [ teams, setTeams ] = useState([]);
    const { id } = useParams();

    const loadTeams = async () => {
        const resp = await getTeams();
        setTeams(resp);
      };

    useEffect( () => {
            loadTeams();
        }, [] )

    const handleSubmit = async (e) => {
        e.preventDefault();
        const resp = await updatePlayerById( id,  { name, position, teamId});
         history.push(`/players/${resp[0].id}`)
    }
    return (
        <>
         <fieldset>
        <legend>Edit Player </legend>     
        <form onSubmit={handleSubmit}>
            <label htmlFor='name' >Name:</label>
            <input id='name' name='name' type="text" value={name} onChange={(e) => setName(e.target.value)} />
            <label htmlFor='position' >position:</label>
            <input id='position' name='position' type="text" value={position} onChange={(e) => setPosition(e.target.value)} />
            <label htmlFor='teamId' >teamId:</label>
            <select id='teamId' name='teamId' onChange={(e) => setTeamId(e.target.value)} >
                    {teams.map((team) => <option key={team.id} value={team.id}>{team.name}</option>)}
            </select>
            
           
            <button type='submit'>Edit Player </button>
        </form>
        </fieldset>   
        </>
    )
}
