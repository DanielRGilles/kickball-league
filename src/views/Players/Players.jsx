import React from 'react'
import { useState, useEffect } from 'react'
import { getPlayers } from '../../services/players.jsx'
import { Link } from 'react-router-dom'


export default function Players() {
    const [ players, setPlayers ] = useState([])
    const [ loading, setLoading ] = useState(true)

    useEffect(() => {
        getPlayers()
        .then((data) => setPlayers(data))
        .catch((error) => console.error(error))
        .finally(() => setLoading(false))
    }, [] )
    
    if (loading) return <h1>Loading players...</h1>
    return (
        <ul>
            {players.map((player) => (
                <li key={player.id}><Link to={`/players/${player.id}`}>{player.name}</Link></li>
            ))}
            {console.log(players)}
        </ul>
    )
}
