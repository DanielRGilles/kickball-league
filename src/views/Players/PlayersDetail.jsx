import React from 'react'

export default function PlayersDetail() {
    return (
        <div>
            
        </div>
    )
}
