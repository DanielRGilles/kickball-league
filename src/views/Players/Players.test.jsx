import { render } from '@testing-library/react'

import Players from './Players'

it('Should fetch then render a list of Players', async () => {
  const { container } = await render(<Players />)
  expect(container).toMatchSnapshot()
})
