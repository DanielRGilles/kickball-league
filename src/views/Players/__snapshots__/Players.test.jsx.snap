// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should fetch then render a list of Players 1`] = `
<div>
  <h1>
    Loading players...
  </h1>
</div>
`;
